github-pullrequest-plugin/src/main/java/org/jenkinsci/plugins/github/pullrequest/trigger/check/PullRequestToCauseConverter.java

```diff
@@ -42,12 +42,18 @@ public static PullRequestToCauseConverter toGitHubPRCause(GitHubPRRepository loc
         return new PullRequestToCauseConverter(localRepo, listener, trigger);
     }
 
+    /**
+     * @return only real trigger cause if matched trigger (not skip) event found for this remotePr.
+     */
     @Override
     public GitHubPRCause apply(final GHPullRequest remotePR) {
-        return from(trigger.getEvents())
+        final GitHubPRCause gitHubPRCause = from(trigger.getEvents())
                 .transform(toCause(remotePR))
                 .filter(notNull())
-                .firstMatch(new SkippedCauseFilter(listener)).orNull();
+                .first()
+                .get();
+
+        return gitHubPRCause.isSkip() ? null : gitHubPRCause;
```

Comment:
> Since FluentIterable doesn't have a reduce method, the only way I was able to get this working by those rules was to unwrap the fluent chain into a regular for(...) loop and break out at first match.

In TODO migrate to java8. Do you know whether it has suitable methods?


> Won't the call to get() here throw an exception if no triggers match (when all events evaluate to a null cause)?

Sounds like we need one more test case :)